From 65dabd6eaf24bbf3b7069a51984411aea2864856 Mon Sep 17 00:00:00 2001
From: Michael Karcher <github@mkarcher.dialup.fu-berlin.de>
Date: Sun, 23 Dec 2018 20:33:52 +0100
Subject: [PATCH] librustc_codegen_llvm: Don't eliminate empty structs in C ABI
 on linux-sparc64

This is in accordance with the SPARC Compliance Definition 2.4.1,
Page 3P-12. It says that structs of up to 8 bytes (which applies
to empty structs as well) are to be passed in one register.
---
 src/librustc_codegen_llvm/abi.rs | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

Index: rustc/src/librustc_codegen_llvm/abi.rs
===================================================================
--- rustc.orig/src/librustc_codegen_llvm/abi.rs
+++ rustc/src/librustc_codegen_llvm/abi.rs
@@ -449,6 +449,9 @@ impl<'tcx> FnTypeExt<'tcx> for FnType<'t
         let linux_sparc64 = target.target_os == "linux"
                        && target.arch == "sparc64"
                        && target.target_env == "gnu";
+        let linux_sparc64 = target.target_os == "linux"
+                       && target.arch == "sparc64"
+                       && target.target_env == "gnu";
         let rust_abi = match sig.abi {
             RustIntrinsic | PlatformIntrinsic | Rust | RustCall => true,
             _ => false
